classes/Group.php

```diff
@@ -109,15 +109,17 @@ public function __construct($id = null, $idLang = null, $idShop = null)
     public static function getGroups($idLang, $idShop = false)
     {
         $shopCriteria = '';
-        if ($idShop) {
+        if(is_int($idShop)){
+            $shopCriteria = ' INNER JOIN '._DB_PREFIX_.'group_shop group_shop ON (group_shop.id_group = g.id_group AND group_shop.id_shop = '.$idShop.')';
+        }elseif ($idShop) {
             $shopCriteria = Shop::addSqlAssociation('group', 'g');
         }
 
         return Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS(
             (new DbQuery())
                 ->select('DISTINCT g.`id_group`, g.`reduction`, g.`price_display_method`, gl.`name`')
                 ->from('group', 'g')
-                ->leftJoin('group_lang', 'gl', 'g.`id_group` = gl.`id_group` AND gl.`id_lang` = '.(int) $idLang)
+                ->leftJoin('group_lang', 'gl', 'g.`id_group` = gl.`id_group` AND gl.`id_lang` = '.(int) $idLang.$shopCriteria)
```

Comment:
will the `->leftJoin(....)` work correctly with
```
$shopCriteria = ' INNER JOIN '._DB_PREFIX_.'group_shop group_shop ON (group_shop.id_group = g.id_group AND group_shop.id_shop = '.$idShop.')';
```
? 
I don't think so. the resulting sql would look something like this:

```sql
SELECT DISTINCT g.`id_group`, g.`reduction`, g.`price_display_method`, gl.`name`
FROM tb_group g
LEFT JOIN tb_group_lang gl ON (g.`id_group` = gl.`id_group` AND gl.`id_lang` = 1 INNER JOIN tb_group_shop group_shop ON (group_shop.id_group = g.id_group AND group_shop.id_shop = 1))
```

Reply:
Yes, it will work. 
I've dumped the resulting SQL it looks like this:

`SELECT DISTINCT g.id_group, g.reduction, g.price_display_method, gl.name FROM tb_group g LEFT JOIN tb_group_lang gl ON g.id_group = gl.id_group AND gl.id_lang = 1 INNER JOIN tb_group_shop group_shop ON (group_shop.id_group = g.id_group AND group_shop.id_shop = 2) ORDER BY g.id_group ASC`

So there are no brakets around the ON Block from the original LEFT JOIN:

`->leftJoin('group_lang', 'gl', 'g.id_group = gl.id_group AND gl.id_lang = '.(int) $idLang)`

to add the brakets it has to be:

`->leftJoin('group_lang', 'gl', '`**(**`g.id_group = gl.id_group AND gl.id_lang = '.(int) $idLang`**.')'**`)`

in this case i would have to change the other line to:

`$shopCriteria = ') INNER JOIN '._DB_PREFIX_.'group_shop group_shop ON (group_shop.id_group = g.id_group AND group_shop.id_shop = '.$idShop.')';` 

But with keeping the original line it will work this way.

BTW: 
The INNER Join is exactly the same, which you get in return from the `Shop::addSqlAssociation('group', 'g')` Method, except that i've changed the use of the  ID from context to the $idShop Parameter.